### PR TITLE
feature: ZENKO-1383 transtion integration tests

### PR DESCRIPTION
**Pull request by bennettbuchanan**
* Add basic lifecycle transition integration tests.
* Upgrade aws-sdk to include lifecycle date field bugfix. See: https://github.com/aws/aws-sdk-js/blob/HEAD/CHANGELOG.md#2600

Note:
These tests will not work until remaining feature PRs are merged.

### CODE REVIEW

File: tests/zenko_tests/node_tests/package.json
Comment (ssalaues) [inline tests/zenko_tests/node_tests/package.json:27]: you will also need to add this to the test entrypoint so that it is ran in the CI  
https://github.com/scality/Zenko/blob/development/1.1/tests/zenko_tests/docker-entrypoint.sh#L27

Comment (bennettbuchanan) [reply tests/zenko_tests/node_tests/package.json:27]: I'm holding off on CI runs because the full feature hasn't been merged yet (I've been using these tests for test-driven-development with our development branches). I'm hoping with this PR to get high-level reviews, and then once all the feature code is merged, we'll be able to enable the CI to help with long-term maintenance.

File: tests/zenko_tests/node_tests/package.json
Comment (jonathan-gramain) [inline tests/zenko_tests/node_tests/package.json:16]: Just out of curiosity, why the change in version?

File: tests/zenko_tests/node_tests/backbeat/LifecycleUtility.js
Comment (jonathan-gramain) [inline tests/zenko_tests/node_tests/backbeat/LifecycleUtility.js]: I think you can use `ReplicationUtility.deleteAllVersions()` even if the bucket is not versioned.

File: tests/zenko_tests/node_tests/backbeat/LifecycleUtility.js
Comment (jonathan-gramain) [inline tests/zenko_tests/node_tests/backbeat/LifecycleUtility.js]: IMO the GET should be done *before* the transition occurs, otherwise we're not testing that transition did not alter the data, we're only testing that a GET on cloudserver returns the same content than a GET directly on the cloud (so testing the GET abstraction layer).

File: tests/zenko_tests/node_tests/package.json
Comment (bennettbuchanan) [reply tests/zenko_tests/node_tests/package.json:16]: There is a bug in the SDK prior to this version that does not handle the Date field correctly. See: https://github.com/aws/aws-sdk-js/blob/HEAD/CHANGELOG.md#2600

File: tests/zenko_tests/node_tests/backbeat/LifecycleUtility.js
Comment (bennettbuchanan) [reply tests/zenko_tests/node_tests/backbeat/LifecycleUtility.js]: Thanks! Nice catch there. I've updated the sequence of events to first get the source object before the lifecycle config is set. I think it ends up with a cleaner test suite in the end, too. 😃

File: tests/zenko_tests/node_tests/backbeat/LifecycleUtility.js
Comment (bennettbuchanan) [reply tests/zenko_tests/node_tests/backbeat/LifecycleUtility.js]: Indeed I can! Updated.

**Review by philipyoo — APPROVED** [review]

File: tests/zenko_tests/node_tests/package.json
Comment (jonathan-gramain) [reply tests/zenko_tests/node_tests/package.json:16]: I see, I think @dora-korpar may have hit this issue a while ago when she was looking at integration tests for lifecycle expiration that were failing for a weird date format issue.

**Review by jonathan-gramain — APPROVED** [review]

File: tests/zenko_tests/node_tests/package.json
Comment (bennettbuchanan) [reply tests/zenko_tests/node_tests/package.json:16]: Yes, it was very confusing to see that.

**Review by ssalaues — APPROVED** [review]

**Comment by bennettbuchanan** [comment]
@bert-e approve

**Comment by bennettbuchanan** [comment]
@bert-e help

**Pull request merged**
